### ISSUE TITLE
frontend: Labels widget

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh-webui-frontend#1138]
### Type of issue
Feature request
### Current behavior
Currently, when adding Sub-Data References to a single entry, this can only be done via included tabs w/ grid view and advanced edit. This is annoying when the data in the included tab only has one column (or a few columns) or is some kind of reference list.
### Expected behavior
We shall have a Lookup which allows opening a selection list (look & feel identical to lookup autoselect). The list opens and the user can scroll through the list, after selection the entry is shown as "ticked", the user can select multiple entries. After closing the selection list, the selected entries are shown as kind of labels (with a circled x to delete the label).

This kind of Lookup can be seen in GitHub Issues as example
![image](https://user-images.githubusercontent.com/13365687/29316008-77ed3282-81c6-11e7-906b-cfbbf3e2bbd1.png)

![image](https://user-images.githubusercontent.com/13365687/29316091-d321b65a-81c6-11e7-83a8-9479e14cb302.png)

**Comment by metas-mk** [comment]
The Design of the labels shall be similar to the Filter Labels set from referenced documents.
![image](https://user-images.githubusercontent.com/13365687/29316477-82044dee-81c8-11e7-84a9-c7f1e965f9d6.png)

**Comment by teosarca** [comment]
note: we have to prepare the API first

**Comment by teosarca** [comment]
API and test case ready.

Open a business partner, e.g. https://w101.metasfresh.com:8443/window/123/2156425

### Layout
![image](https://user-images.githubusercontent.com/1244701/30681342-b32e95da-9ead-11e7-94af-e7c8a2dcabaa.png)

### Fetching available labels
To fetch ALL available labels (already assigned or not) you can use:
* standard dropdown endpoint to get all values. e.g. http://w101.metasfresh.com:8081/rest/api/window/123/2156425/field/Labels_548674/dropdown
* standard typeahead endpoint to fetch those values which contain a given string. e.g. http://w101.metasfresh.com:8081/rest/api/window/123/2156425/field/Labels_548674/typeahead?query=Beh

See
* http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldDropdownUsingGET
* http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldDropdownUsingGET_1
* http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldTypeaheadUsingGET
* http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentFieldTypeaheadUsingGET_1


### Field value
The field value can be obtained and it's handled like any other widget, using exactly the same API.
The value format is:
```
{
	"values": [
		{
			"K_Firma": "Firma"
		},
		{
			"K_Behörden": "Behörden"
		}
	]
}
```
Basically, an array of labels (described by key/name pairs)

Those labels are the assigned labels.
If you want to add or remove or change assigned labels, you just need to change the array and PATCH the document, same like any other field.
e.g.
```
curl -X PATCH --header 'Content-Type: application/json' --header 'Accept: application/json' -d '[ \ 
   { \ 
     "op": "replace", \ 
     "path": "Labels_548674", \ 
     "value": {"values": [{"K_Firma": "Firma"}, {"K_Behörden": "Behörden"}]} \ 
   } \ 
 ] \ 
 ' 'http://w101.metasfresh.com:8081/rest/api/window/123/2156425?advanced=false'
```

**Comment by teosarca** [comment]
awesome, quick tested and integrated.

**Comment by metas-rc** [comment]
Results of IT

* Adding and removing labels works correctly in the window Business Partner
